src/session_impl.cpp

```diff
@@ -3145,7 +3133,8 @@ namespace aux {
 			for (auto& ext : m_ses_extensions)
```

Comment:
would it make more sense to have separate lists for separate features? the "last minute check" is a bit of an anti-pattern